src/platform/Kconfig

```diff
@@ -353,7 +353,7 @@ config CAVS
 	select INTEL
 	select XT_INTERRUPT_LEVEL_2
 	select XT_INTERRUPT_LEVEL_5
-	select INTEL_HDA
+	select INTEL_HDA if !ZEPHYR_NATIVE_DRIVERS
```

Comment:
does this and the changes below mean that native builds are broken before this commit? Are they broken before this series too? If they are, then it doesn't matter a lot in which order we fix the breakage. If it isn't currently broken, then it would be good to order and group these commits in a way to avoid regressions between commits too.